src/handlers/handlers.c

```diff
@@ -23,6 +24,20 @@ uint16_t handleApdu(command_t *cmd, bool isNewCall) {
         return ERR_BAD_CLA;
     }
 
+#ifdef HAVE_SWAP
+    if (G_called_from_swap) {
+        if ((cmd->ins != INS_GET_PUBLIC_KEY) && (cmd->ins != INS_DERIVE_ADDRESS) &&
+            (cmd->ins != INS_GET_VERSION) && (cmd->ins != INS_SIGN_TX)) {
+            PRINTF("Refused INS when in SWAP mode\n");
+            return ERR_UNKNOWN_INS;
+        }
+        if ((cmd->ins == INS_DERIVE_ADDRESS) && (cmd->p1 != 0x01)) {
```

Comment:
Maybe also block INS_GET_PUBLIC_KEY if in UI is asked
Not sure if cardano has both the no ui and the ui INS_GET_PUBLIC_KEY

Reply:
INS_GET_PUBLIC_KEY is also called during swap from LL, and the UI is blocked in the dedicated code